Add changelog for memgraph lab 2.4.0 (#628)
memgraph/docs: lab/changelog.md

sidebar_label: Changelog
---

## v2.4.0 - Dec 2, 2022

## What's New

> * Support for manual transaction workflow has been enabled using transaction commands `BEGIN`, `COMMIT`, and `ROLLBACK`.

Reviewer: ```suggestion
* Memgraph Lab now supports manual transaction workflow you can construct using transaction commands `BEGIN`, `COMMIT`, and `ROLLBACK`.
```

* Cypher intellisense has been updated to suggest new Cypher features from Memgraph 2.4.0 such as:
  * Privileges for user-role authorization.
  * Commands and privileges for label-based authorization.
  * Manual transaction commands: `BEGIN`, `COMMIT`, `ROLLBACK`.
  * Checking configuration with `SHOW CONFIG`.
  * All shortest path algorithm `allShortest`.
  * Graph projection function `project`.
  * Additional query module signature that accepts a projected graph as an optional first argument.
* Graph results view will check for nodes and relationships in arrays and projected graphs. It simplifies
  the visualization of a projected graph or an array of nodes/relationships without using `UNWIND`.

## Bug Fixes

> * Once the table results view is selected, an automatic switch to the graph view won't happen on the next query run.

Reviewer: ```suggestion
* Once the table results view is selected, the results of the following query run will also preview in the table results view, instead of automatically switching to the graph view.
```

> * Exploring a dataset's query collection now works as expected. It opens up a query list to explore the dataset.

Reviewer: ```suggestion
* Exploring a dataset's query collection now works as expected. It opens up a list of queries that can be used to explore the dataset.
```

> * Missing detailed error message on a failed query in rich collections has been fixed.

Reviewer: ```suggestion
* Failed queries from the rich collections now return a detailed error message.
```

was this the point of this fix?

Author: Before, you would just get "Query failed" in the rich collection and that's it. Now it returns "what actually went wrong with the query" so you can fix it.

> * Save code changes action in rich collections will be enabled only if there are unsaved changes for the Cypher query 
>   or GSS.

Reviewer: ```suggestion
* _Save code changes_ action in rich collections will now be enabled only if there are unsaved changes for the Cypher query 
  or GSS.
```

> * A bug in the table results view where only the first node label has been shown instead of all node labels has been fixed.

Reviewer: ```suggestion
* A bug that would only show the first node label instead of all node labels in the table results view has been fixed.
```
